src/OpenTelemetry.Api/Trace/ActivityExtensions.cs

```diff
@@ -69,14 +69,26 @@ public static Status GetStatus(this Activity activity)
         }
 
         /// <summary>
-        /// Record Exception.
+        /// Adds an activity event containing information from the specified exception.
         /// </summary>
         /// <param name="activity">Activity instance.</param>
         /// <param name="ex">Exception to be recorded.</param>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static void RecordException(this Activity activity, Exception ex)
         {
-            if (ex == null)
+            activity?.RecordException(ex, default);
+        }
+
+        /// <summary>
+        /// Adds an activity event containing information from the specified exception and additional tags.
+        /// </summary>
+        /// <param name="activity">Activity instance.</param>
+        /// <param name="ex">Exception to be recorded.</param>
+        /// <param name="tags">Additional tags to record on the event.</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static void RecordException(this Activity activity, Exception ex, in TagList tags)
```

Comment:
unless Activity.AddEvent/team also accepts TagList, the benefits of using TagList in this API is nil, as we anyway have to allocate the ActivityTagsCollection.

Hopefully if/when the spec is stable, we can bring this up.

Reply:
I think the idea is that the consumer can still avoid some allocations when using `TagList`, but yea the benefit would be greater if `AddEvent` accepted TagList as well.